### PR TITLE
feat: pnpm

### PR DESCRIPTION
**Pull request by tyrauber**
Remove yarn / pnp, in favor of pnpm.

 ✅ $ pnpm unittest
 ✅ $ pnpm test:plugin
 ✅ $ pnpm build:plugin
 ✅ $ pnpm generate
 🚩 $ pnpm lint

## Checklist

- [ ] I have tested this on a device/simulator for each compatible OS
- [ ] I formatted JS and TS files with running `yarn lint:fix` in the root folder
- [X] I have run tests via `yarn test` in the root folder
- [X] I updated the documentation with running `yarn generate` in the root folder
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typings files (`index.d.ts`)
- [ ] I added/updated a sample (`/example`)

### CODE REVIEW

**Comment by sarahsporck** [comment]
What would be tha advantage of pnpm in this repo? Could you work out some more context?

**Comment by tyrauber** [comment]
Hi @sarahsporck, the primary motivation was to reduce the amount of build complexity with the current setup of yarn and pnp.  I believe, and please correct me if I am wrong, but this PR eliminates a whole lot of unnecessary code and configuration  - 45,000 lines.  Please confirm. Tests run, lint works and the plugin and library build.  

There are additional benefits of PNPM over yarn and npm. It works well out of the box with monorepos and is really good at managing dependency disk space.  I might be able to get both yarn and pnpm working in the same repo, if you think that is beneficial.

Take a look at #162 that builds upon this PR to implement Expo in a Monorepo environment.

File: package.json [hunk 1]
```diff
@@ -22,17 +22,17 @@
   },
   "scripts": {
     "fetch:style:spec": "./scripts/download-style-spec.sh",
-    "generate": "yarn node ./scripts/autogenerate",
-    "test": "npm run lint && npm run unittest",
+    "generate": "npx node ./scripts/autogenerate",
```
Comment (sarahsporck) [inline package.json:25]: Why do we need `npx` here?
```suggestion
    "generate": "node ./scripts/autogenerate",
```

File: package.json [hunk 2]
```diff
@@ -22,17 +22,17 @@
...
     "unittest:single": "jest --testNamePattern",
     "lint": "eslint .",
     "lint:fix": "eslint . --fix",
-    "prepack": "pinst --disable && yarn build:plugin",
+    "prepack": "pinst --disable && pnpm build:plugin",
     "test:plugin": "jest plugin",
     "build:plugin": "tsc --build plugin",
     "lint:plugin": "eslint plugin/src/*",
-    "postinstall": "husky install",
+    "_postinstall": "husky install",
```
Comment (sarahsporck) [inline package.json:35]: Doing this in postinstall was a yarn 2+ workaround. Look here how to properly configure husky for pnpm https://typicode.github.io/husky/getting-started.html#automatic-recommended

**Review by sarahsporck — CHANGES_REQUESTED** [review]
Could you run pnpm install in the example repo, too? It also seems there is a missing peer dependency here. I guess it could also make sense to set up the example repo as a workspace, but I'll keep that for you to decide.

**Pull request closed without merging**